tests/test_exceptions.py

```diff
@@ -76,7 +76,7 @@ def test_cross_module_exceptions(msg):
 
 # TODO: FIXME
 @pytest.mark.xfail(
-    "env.MACOS and env.PYPY",
+    "(env.MACOS and env.PYPY) or env.ANDROID",
```

Comment:
I don't understand what's going on here, but going by the linked issues and PRs, neither do the pybind11 maintainers, so I don't feel too bad.